packages/components/addon/components/hds/table/index.js

```diff
@@ -185,85 +184,83 @@ export default class HdsTableIndexComponent extends Component {
   }
 
   @action
-  onSelectionAllChange(headerCheckbox) {
-    const checkboxes = this.allTbodyCheckboxes();
-    this.setHeaderCheckboxAriaLabel(headerCheckbox);
-    checkboxes.forEach((checkbox) => {
-      checkbox.checked = headerCheckbox.checked;
-      this.setRowCheckboxAriaLabel(checkbox);
+  onSelectionAllChange() {
+    // this.setSelectAllCheckboxAriaLabel(this.selectAllCheckbox);
+    this.selectableRows.forEach((row) => {
+      row.checkbox.checked = this.selectAllCheckbox.checked;
+      // this.setSelectRowCheckboxAriaLabel(row.checkbox);
     });
+    this.isSelectAllCheckboxSelected = this.selectAllCheckbox.checked;
 
     let { onSelectionChange } = this.args;
     if (typeof onSelectionChange === 'function') {
-      onSelectionChange(this.selectionKeys);
+      onSelectionChange({
+        selectionKey: 'all',
+        selectionCheckboxElement: this.selectAllCheckbox,
+        selectableRows: this.selectableRows,
+        selectableRowsStates: this.selectableRows.reduce((acc, row) => {
+          acc[row.selectionKey] = row.checkbox.checked;
+          return acc;
+        }, {}),
+      });
```

Comment:
This seems a good bit more complex. What are the advantages of passing back all these different items?

Reply:
Discussed in person: the reason for this is to provide consumers different options for what they may want to use as argument for their callbacks. To be validated with consumers/ambassadors before releasing.